clang/test/SemaCXX/warn-thread-safety-analysis.cpp

```diff
@@ -1718,6 +1720,13 @@ struct TestScopedLockable {
     MutexLock{&mu1}, a = 5;
   }
 
+  void temporary2(int x) {
+    if (check(MutexLock{&mu1}) || x) {
+
+    }
```

Comment:
The `if` is probably not needed here, right? We could just write
```suggestion
    check(MutexLock{&mu1}) || x;
```
and should have the same CFG artifact.